src/main/java/com/zaxxer/nuprocess/linux/ProcessEpoll.java

```diff
@@ -245,28 +251,25 @@ else if (ident == stdinFd) {
       }
       finally {
          if (linuxProcess != null) {
-            if (stdinFd != null) {
+            if (stdinFd != Integer.MIN_VALUE) {
                linuxProcess.getStdin().release();
             }
-            if (stdoutFd != null) {
+            if (stdoutFd != Integer.MIN_VALUE) {
                linuxProcess.getStdout().release();
             }
-            if (stderrFd != null) {
+            if (stderrFd != Integer.MIN_VALUE) {
                linuxProcess.getStderr().release();
             }
          }
-         triggeredEvent.clear();
```

Comment:
This is a "cleaner" approach, but the contents of `triggeredEvent` at the end of the call don't matter. It's guaranteed the next call will be to `epoll_wait`, which will not read those contents--it'll only overwrite them if a descriptor is ready.